### PR TITLE
Wire up CPU options

### PR DESCRIPTION
**Pull request by inlined**
Works for everything but 128MB functions with `gcf_gen1` CPU because we actually need 128MiB functions to qualify for that RAM amount. It doesn't seem like 128MB even works.

### CODE REVIEW

**Review by taeold — COMMENTED** [review]
I'll review this PR once we get approval to move to MiB (vs MB)?

File: src/deploy/functions/release/fabricator.ts
Comment (taeold) [inline src/deploy/functions/release/fabricator.ts:341]: having a brain fart, but should this be comparing the `endpoint.cpu` to `memoryToGen2Cpu` instead?

In english words, I would've guessed that:

```
hasCustomCPU = endpoint.cpu does not match default settings for Gen 2 memory -> CPU mapping.
```

File: src/deploy/functions/release/fabricator.ts
Comment (taeold) [inline src/deploy/functions/release/fabricator.ts:446]: nit* `hasCustomCPU` for consistency with above.

File: src/deploy/functions/release/fabricator.ts
Comment (taeold) [inline src/deploy/functions/release/fabricator.ts:677]:
```suggestion
    `Service ${service.metadata.name} has unexpected status ${JSON.stringify(
```

File: src/deploy/functions/release/fabricator.ts
Comment (taeold) [inline src/deploy/functions/release/fabricator.ts:526]: Isn't this an infinite loop? or are we expecting `serviceIsResolved` to throw an error at some point?

File: src/gcp/run.ts
Comment (taeold) [inline src/gcp/run.ts:131]: nit* couple of unexpectedly empty docstring comments here and below.

File: src/test/deploy/functions/release/fabricator.spec.ts
Comment (taeold) [inline src/test/deploy/functions/release/fabricator.spec.ts:1015]:
```suggestion
    it("returns false if the observed generation isn't the metageneration", () => {
```

File: src/test/deploy/functions/release/fabricator.spec.ts
Comment (taeold) [inline src/test/deploy/functions/release/fabricator.spec.ts:1046]: the test description and expectatoin doesn't seem to match here.

**Review by taeold — COMMENTED** [review]
overall looks great, some comments to consider b/4 merging.

File: src/deploy/functions/prepare.ts
Comment (inlined) [reply src/deploy/functions/prepare.ts]: Can't remove; linter just adds it back. Will just add a comment.

File: src/deploy/functions/release/fabricator.ts
Comment (inlined) [reply src/deploy/functions/release/fabricator.ts:345]: I can change the name to be more clear or add a comment. This code is correct; gcf gen 1 is the "built in" memory that we get automatically and v2 is a default we use for our custom VM shapes.

File: src/deploy/functions/release/fabricator.ts
Comment (inlined) [reply src/deploy/functions/release/fabricator.ts]: Sure

File: src/deploy/functions/release/fabricator.ts
Comment (inlined) [reply src/deploy/functions/release/fabricator.ts:530]: serviceIsResolved returns true once observedGeneration == spec.generation and Condition[Type=Ready].State is True

File: src/deploy/functions/release/fabricator.ts
Comment (inlined) [reply src/deploy/functions/release/fabricator.ts:681]: We're specifically talking about the Status of Type=Ready so I think the comment is correct.

File: src/gcp/run.ts
Comment (inlined) [reply src/gcp/run.ts]: Will just document

File: src/test/deploy/functions/release/fabricator.spec.ts
Comment (inlined) [reply src/test/deploy/functions/release/fabricator.spec.ts]: Odd... That shouldn't pass. I'll dive deeper.

**Review by taeold — APPROVED** [review]

**Pull request merged**
